library/Core.cpp

```diff
@@ -512,6 +512,36 @@ static bool try_autocomplete(color_ostream &con, const std::string &first, std::
     return false;
 }
 
+static CommandList get_all_commands()
+{
+    CommandList commands;
+    for (auto const &command : built_in_commands)
+            commands.insert(command);
+    
+    auto plug_mgr = Core::getInstance().getPluginManager();
+    for (auto it = plug_mgr->begin(); it != plug_mgr->end(); ++it)
+    {
+        const Plugin *plug = it->second;
+        for (size_t j = 0; j < plug->size(); j++)
+        {
+            const PluginCommand &pcmd = (*plug)[j];
+            if (!pcmd.isHotkeyCommand())
+            {
+                commands.insert(pcmd.name);
+            }
+        }
+    }
+    
+    std::map<string, string> scripts;
+    listAllScripts(scripts, true);
+    for (auto iter = scripts.begin(); iter != scripts.end(); ++iter)
+    {
+        commands.insert(iter->first);
+    }
+
+    return commands;
```

Comment:
This doesn't appear to include aliases - they probably aren't that commonly-used, but `Core::ListAliases()` appears to be a safe way to get a list of them.

Reply:
I will change that in spare time.